### PR TITLE
modify the return value of NewContextMessageLayer()

### PR DESCRIPTION
**Pull request by zhuguihua**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind bug

**What this PR does / why we need it**:
modify the return value of function NewContextMessageLayer() which has no error.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by anyushun** [comment]
message layer is mandatory, good work
/lgtm

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 1]
```diff
@@ -1095,6 +1095,7 @@ func NewUpstreamController() (*UpstreamController, error) {
 	ml, err := messagelayer.NewContextMessageLayer()
```
Comment (kadisi) [inline cloud/pkg/edgecontroller/controller/upstream.go]: NewContextMessageLayer() function only returns an `ContextMessageLayer object`  and the err always is nil.
I suggest modifying the return value of this function。

`func NewContextMessageLayer() (MessageLayer, error)` -> `func NewContextMessageLayer() (MessageLayer)`

Comment (fisherxu) [reply cloud/pkg/edgecontroller/controller/upstream.go]: Sounds good.

Comment (zhuguihua) [reply cloud/pkg/edgecontroller/controller/upstream.go]: agreed, updated.

File: cloud/pkg/devicecontroller/controller/upstream.go [hunk 2]
```diff
@@ -189,11 +189,19 @@ func (uc *UpstreamController) unmarshalDeviceStatusMessage(msg model.Message) (*
 // NewUpstreamController create UpstreamController from config
 func NewUpstreamController(dc *DownstreamController) (*UpstreamController, error) {
 	config, err := utils.KubeConfig()
+	if err != nil {
+		klog.Warningf("create kube client failed with error: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/controller/upstream.go]:
```suggestion
		klog.Warningf("Failed to create kube client: %s", err)
```

File: cloud/pkg/devicecontroller/controller/upstream.go [hunk 3]
```diff
@@ -189,11 +189,19 @@ func (uc *UpstreamController) unmarshalDeviceStatusMessage(msg model.Message) (*
...
+		return nil, err
+	}
+
 	crdcli, err := utils.NewCRDClient(config)
-	ml, err := messagelayer.NewContextMessageLayer()
 	if err != nil {
-		klog.Warningf("Create message layer failed with error: %s", err)
+		klog.Warningf("create crd client failed with error: %s", err)
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/controller/upstream.go]:
```suggestion
		klog.Warningf("Failed to create crd client: %s", err)
```

File: cloud/pkg/devicecontroller/controller/upstream.go [hunk 4]
```diff
@@ -189,11 +189,19 @@ func (uc *UpstreamController) unmarshalDeviceStatusMessage(msg model.Message) (*
...
-		klog.Warningf("Create message layer failed with error: %s", err)
+		klog.Warningf("Failed to create crd client: %s", err)
+		return nil, err
 	}
+
+	ml := messagelayer.NewContextMessageLayer()
+
 	uc := &UpstreamController{
 		crdClient:    crdcli,
 		messageLayer: ml,
```
Comment (kadisi) [inline cloud/pkg/devicecontroller/controller/upstream.go]: use messagelayer.NewContextMessageLayer() directly?

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 5]
```diff
@@ -1092,10 +1092,7 @@ func NewUpstreamController() (*UpstreamController, error) {
...
 		return nil, err
 	}
-	ml, err := messagelayer.NewContextMessageLayer()
-	if err != nil {
-		klog.Warningf("create message layer failed with error: %s", err)
-	}
+	ml := messagelayer.NewContextMessageLayer()
 	uc := &UpstreamController{
 		kubeClient:   cli,
 		messageLayer: ml,
```
Comment (kadisi) [inline cloud/pkg/edgecontroller/controller/upstream.go]: use messagelayer.NewContextMessageLayer()?

Comment (zhuguihua) [reply cloud/pkg/edgecontroller/controller/upstream.go]: updated.

**Pull request merged**